File: py/Task69.py
# 实现 int sqrt(int x) 函数。

# 计算并返回 x 的平方根，其中 x 是非负整数。

# 由于返回类型是整数，结果只保留整数的部分，小数部分将被舍去。

# 示例 1:

# 输入: 4
# 输出: 2
# 示例 2:

# 输入: 8
# 输出: 2
# 说明: 8 的平方根是 2.82842...,
#      由于返回类型是整数，小数部分将被舍去。


class Solution:
    def mySqrt(self, x: int) -> int:
        if not x:
            return 0
        left, right = 0, x
        while left <= right:
            mid = (left + right) >> 1
            if mid ** 2 > x:
                right = mid - 1
            else:
                left = mid + 1
        return right


x = 8
obj = Solution()
print(obj.mySqrt(x))

File: java/leetcode/src/Task69.java
import org.junit.jupiter.api.Test;

public class Task69 {

    public int mySqrt(int x) {
        int left = 1, right = x;
        int res=0;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            long result = 1L* mid * mid;
            if (result == x) {
                return mid;
            } else if (result > x) {
                right = mid - 1;

            } else {
                left = mid+1;
                res = mid;
            }

        }
        return res;
    }

    @Test
    void test(){
        System.out.println(mySqrt(2147395599));
    }
}
